### PR TITLE
Add Coverage For Save Method In WhitelistSupportImplTest

### PR DESCRIPTION
**Pull request by wilmerrootstock**
## Description
A new `save` method was added to WhitelistSupport, so we must add coverage.

## Motivation and Context
This relates to the Bridge Refactor to decouple some objects tied closely to the Bridge.

## How Has This Been Tested?
Unit Tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java [hunk 1]
```diff
@@ -104,6 +109,19 @@ void getLockWhitelistEntryByAddress_whenLockWhitelistHasEntries_shouldReturnOneO
         assertEquals(btcAddress, actualLockWhitelistEntry.address());
     }
 
+    private void saveInMemoryStorageOneOffWhiteListEntry() {
+        Coin maxTransferValue = Coin.COIN;
+        OneOffWhiteListEntry oneOffWhiteListEntry = new OneOffWhiteListEntry(btcAddress, maxTransferValue);
+        List<OneOffWhiteListEntry> oneOffWhiteListEntries = Collections.singletonList(oneOffWhiteListEntry);
+        Pair<List<OneOffWhiteListEntry>, Integer> pairValue = Pair.of(oneOffWhiteListEntries, 100);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java:116]: What is `100`? Maybe put in a variable so it's easy to understand what it represents

File: rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java [hunk 2]
```diff
@@ -293,16 +311,58 @@ void verifyLockSenderIsWhitelisted_whenAddressIsNotWhitelisted_shouldReturnFalse
...
-        );
+        int actualSize = whitelistSupport.getLockWhitelistSize();
+        assertEquals(0, actualSize);
+        assertNull(whitelistSupport.getLockWhitelistEntryByIndex(0));
+        assertNull(whitelistSupport.getLockWhitelistEntryByIndex(1));
+    }
+
+    @Test
+    void save_whenRSKIP87IsNotActive_shouldSaveOneOffLockWhitelistAddress() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(false);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java:326]: Let's use ActivationConfigForTest class as discussed

File: rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java [hunk 3]
```diff
@@ -293,16 +311,58 @@ void verifyLockSenderIsWhitelisted_whenAddressIsNotWhitelisted_shouldReturnFalse
...
+    @Test
+    void save_whenRSKIP87IsActive_shouldSaveUnlimitedLockWhitelistAddress() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(true);
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, WhitelistCaller.AUTHORIZED.getRskAddress());
+        whitelistSupport.addUnlimitedLockWhitelistAddress(tx, btcAddress.toString());
+
+        whitelistSupport.save();
+
+        int actualSize = whitelistSupport.getLockWhitelistSize();
+        Address actualBtcAddress = whitelistSupport.getLockWhitelistEntryByAddress(btcAddress.toString()).address();
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java:347]: This test and the one above seem to be basically the same. There is no assertion of what was saved, it only changes the method you call in WhitelistSupport. But doesn't seem to have any relationship with the `save` method right?

Comment (wilmerrootstock) [reply rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java:347]: Actually, they have a direct relation with the `save` method. At first, they use `addUnlimitedLockWhitelistAddress` and `addOneOffLockWhitelistAddress` to add an address to be whitelisted, and then the methods save those addresses in storage with the `save` method.

In the assertion part, you can see we guarantee the address was saved by getting the `LockWhitelistSize` from the storage and getting the address we saved with the `getLockWhitelistEntryByAddress` method directly from the storage as well.

File: rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java (hunk 1 above)
Comment (wilmerrootstock) [reply rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java]: Done! The Magic number was replaced with a meaningful constant.

File: rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java (hunk 2 above)
Comment (wilmerrootstock) [reply rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistSupportImplTest.java]: Done! I uploaded it in a separate commit. Please ensure these are all the changes required.

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**